autorest/codegen/serializers/model_python3_serializer.py

```diff
@@ -58,7 +58,9 @@ def init_args(model: ObjectSchema) -> List[str]:
                 init_args.append(f"self.{prop.name} = None")
             elif prop.is_discriminator:
                 discriminator_value = f"'{model.discriminator_value}'" if model.discriminator_value else None
-                init_args.append(f"self.{prop.name} = {discriminator_value}")
+                # adding the type ignore because mypy throws an incompatible type error because
+                # the children have a value for the discriminator, while the parent sets it to None
+                init_args.append(f"self.{prop.name} = {discriminator_value}  # type: ignore")
```

Comment:
If you do the parent with `# type: Optional[str]` and the child `# type: str` should be fine? (str is an example, we should use the actual type)

Reply:
that works, implementing that change now. Thanks!